pkg/buildpack/managed_collection.go

```diff
@@ -47,30 +40,17 @@ func (f *ManagedCollection) FlattenedModules() [][]BuildModule {
 	return nil
 }
 
-// AddModules determines whether the explodedModules must be added as flattened or not. It uses
-// flatten and maxDepth configuration given during initialization of the manager.
+// AddModules determines whether the explodedModules must be added as flattened or not.
 func (f *ManagedCollection) AddModules(main BuildModule, deps ...BuildModule) {
 	if !f.flatten {
 		// default behavior
 		f.explodedModules = append(f.explodedModules, append([]BuildModule{main}, deps...)...)
 	} else {
-		if f.maxDepth <= FlattenMaxDepth {
-			// flatten all
-			if len(f.flattenedModules) == 1 {
-				f.flattenedModules[0] = append(f.flattenedModules[0], append([]BuildModule{main}, deps...)...)
-			} else {
-				f.flattenedModules = append(f.flattenedModules, append([]BuildModule{main}, deps...))
-			}
+		// flatten all
+		if len(f.flattenedModules) == 1 {
+			f.flattenedModules[0] = append(f.flattenedModules[0], append([]BuildModule{main}, deps...)...)
```

Comment:
I think you explained this to me before, but what is the difference between the if/else here? (Perhaps a code comment could help future readers)